### ISSUE TITLE
Sony latest firmware bootloop

### ISSUE THREAD

**Issue by itandy** [rovo89/Xposed#339]
I just upgraded my Xperia X Compact (Oreo) from previous version (34.4.A.2.19, Feb patch) to (34.4.A.2.32, March patch). But once I enable Xposed my phone is unable to finish a reboot. Other Xperia users (X Compact and X with March firmware) are having similar problem so it seems to be an compatibility issue between Xposed and our latest ROM. Previous firmware (34.4.A.2.19) works completely fine. Just this minor upgrade caused the problem.

Logcat
https://pastebin.com/0qb5mjfg

**Comment by wanam** [comment]
> 03-18 18:47:14.976 W/zygote64( 4440): Got request to deoptimize un-deoptimizable method void com.android.server.SystemServer.run()
> 03-18 18:47:14.977 F/zygote64( 4440): quick_exception_handler.cc:326] Check failed: callee_method_ != nullptr 

Same as [this issue](https://github.com/rovo89/Xposed/issues/325).

**Comment by itandy** [comment]
I thought #325 is caused by another package so is a different issue. Please correct me if I'm wrong.

**Comment by wanam** [comment]
The module is just a trigger, you may still get this issue with/out modules enabled.

**Comment by itandy** [comment]
Yes, I got this issue with no modules at all. Since #325 is a Oreo 8.1 issue, does that mean our latest Sony firmware (still 8.0) brings something from 8.1 which causes the same issue with Xposed?

**Comment by rovo89** [comment]
Thanks @wanam for jumping in here. Yes, this sounds a lot like the general issue with the new approach which I'm already aware of. This affects Oreo in general, and would even affect Nougat.

**Comment by itandy** [comment]
Thanks @rovo89! Just curious, why would only this new Sony firmware be affected by this issue and not the previous one? This is the third firmware for Oreo for my device. I thought this minor upgrade is only a security patch together with some minor bug fixes.

**Comment by rovo89** [comment]
This doesn't even need to be a code change. It depends on what machine code exactly is generated from it. Under certain circumstances, it doesn't work well with Xposed's way of invalidating callers of hooked methods. See https://github.com/rovo89/Xposed/issues/325#issuecomment-367012394 for an attempt to explain this complex situation in few words.

**Comment by itandy** [comment]
Thanks again! Seems like we're just in bad luck this time. Anyway, we'll wait for your solution.

**Comment by klcb2010** [comment]
all so Sony xzp by  March patch

**Comment by pqkluan** [comment]
I want to add that my Sony Xperia x F5122 with the latest update (34.4.A.2.50) also has the same issue.

**Comment by Iria-D** [comment]
Xperia XZ Premium also same issue after March security update

**Comment by Iria-D** [comment]
https://forum.xda-developers.com/xz-premium/development/rom-stock-light-rom-v4-debloated-t3813039/page7

It seems beta2 xposed is compatible beta 3 cause bootloop
Edit: some modules not compatible
xprivacylua causes crash, gravitybox too.

**Comment by TempleZhou** [comment]
My Sony Xperia XZ1 with the latest update also has the same issue.

**Comment by omeryagmurlu** [comment]
any updates on this?

**Comment by wwwdesign** [comment]
One year later... Any news?
